pkg/controller/volume/attachdetach/cache/actual_state_of_world.go

```diff
@@ -384,24 +388,15 @@ func (asw *actualStateOfWorld) AddVolumeNode(
 	return volumeName, nil
 }
 
-func (asw *actualStateOfWorld) SetVolumeMountedByNode(
-	logger klog.Logger,
-	volumeName v1.UniqueVolumeName, nodeName types.NodeName, mounted bool) error {
+func (asw *actualStateOfWorld) SetVolumesMountedByNode(
+	logger klog.Logger, volumeNames []v1.UniqueVolumeName, nodeName types.NodeName) {
 	asw.Lock()
 	defer asw.Unlock()
 
-	volumeObj, nodeObj, err := asw.getNodeAndVolume(volumeName, nodeName)
-	if err != nil {
-		return fmt.Errorf("failed to SetVolumeMountedByNode with error: %v", err)
-	}
-
-	nodeObj.mountedByNode = mounted
-	volumeObj.nodesAttachedTo[nodeName] = nodeObj
-	logger.V(4).Info("SetVolumeMountedByNode volume to the node",
+	asw.inUseVolumes[nodeName] = sets.New(volumeNames...)
```

Comment:
This could be for a subsequent PR / different issue: do we ever delete entries in `asw.inUseVolumes`? When a node gets deleted, A/D controller calls SetVolumesMountedByNode with the last known `node.status.volumesInUse`.

If `volumesInUse` is empty, I think we could safely delete `inUseVolumes[nodeName]` here, what do you think?

If `volumesInUse` is not empty, they will stay "mounted" on the node in ASW forever. With potentially many nodes autoscaled / deleted during a cluster lifecycle, we should have a mechanism to delete nodes that never came back. But that's for another discussion/PR.

Reply:
Yes, this is a future work. Can we assume the deleted node is already shutdown/cleaned up? So that we can just detach every volume from it.